### ISSUE TITLE
Can IJSHandle.EvaluateFunctionAsync return null ?

### ISSUE THREAD

**Issue by CaCTuCaTu4ECKuu** [hardkoded/puppeteer-sharp#2805]
I'm trying to use PuppeteerSharp.Dom and I believe it takes parsing approach when it comes to returning generic values as evaluation results compared to PuppeteerSharp which consider original variable type from JS and throws when requesting something like int when evaluating result is not Number so I was updating repo and got a question

[Here ](https://github.com/ChromiumDotNet/PuppeteerSharp.Dom/commit/a7b7b03c1d27ce5754e84fd38a0d637495152904) at line [71](https://github.com/ChromiumDotNet/PuppeteerSharp.Dom/commit/a7b7b03c1d27ce5754e84fd38a0d637495152904#diff-1759b5b2a0691057a56c32be437f98f2ca7bdb19e0cbdd0cec19b78b7abcacddR71) `IJSHandle.EvaluateFunctionAsync` is called which is `Task<JsonElement?> EvaluateFunctionAsync(string script, params object[] args)`
But while I was testing I didn't see result as null a single time. For example getting non-existent smth. returns JsonElement and at that point I'm not sure what then should return null.

Looks like it's a mistake and `JsonElement?` must be changed to just `JsonElement` to remove confusion

And while I'm at it - did I get it right about translating/parsing types from evaluation results to generic types? I run some tests to check
`await page.EvaluateFunctionAsync<int>("() => '123'")` dont work but
`await page.EvaluateFunctionAsync<int>("() => Number('123')")` works

**Comment by kblok** [comment]
I think `JsonElement?` makes sense when the function returns no value at all.

Regarding this:

```
await page.EvaluateFunctionAsync<int>("() => '123'") dont work but
```

That's part of switching to `System.Text.Json` in v19. `System.Text.Json` is more strict than Newtonsoft.

**Comment by CaCTuCaTu4ECKuu** [comment]
Yes it does. I'm mentioning it because I didn't catch an eye on anything that can return null while I was debugging.
But it's a big and complex project that I tbh can't dedicate enough time so sorry for wasting time if I'm wrong.

Basically `JsonValueKind` represent response and have `Undefined` and `Null` so my guess was that if some code that returns `default` and if it was missed during transition result would never be null. Or it was intentional. Or there is a state when it will return null?

**Comment by kblok** [comment]
> Yes it does. I'm mentioning it because I didn't catch an eye on anything that can return null while I was debugging. But it's a big and complex project that I tbh can't dedicate enough time so sorry for wasting time if I'm wrong.
> 
> Basically `JsonValueKind` represent response and have `Undefined` and `Null` so my guess was that if some code that returns `default` and if it was missed during transition from JToken and result would never be null. Or it was intentional. Or there is a state when it will return null?

Do you have any example to look at?

**Comment by CaCTuCaTu4ECKuu** [comment]
I just rerun tests with conditional breakpoints and I didnt catch null result even once
So [here](https://github.com/urgcorp/PuppeteerSharp.Dom/blob/a29fe2cecd3c5e46fc09a74e3c050f8ea96f9b07/PuppeteerSharp.Dom/DomHandle.cs#L220) is method that ensures result parsing and it's called on:
* `EvaluateFunctionInternalAsync<T>` - [line 72](https://github.com/urgcorp/PuppeteerSharp.Dom/blob/a29fe2cecd3c5e46fc09a74e3c050f8ea96f9b07/PuppeteerSharp.Dom/DomHandle.cs#L72)
* `IJSHandle.EvaluateFunctionAsync<T>` - [line 184](https://github.com/urgcorp/PuppeteerSharp.Dom/blob/a29fe2cecd3c5e46fc09a74e3c050f8ea96f9b07/PuppeteerSharp.Dom/DomHandle.cs#L184)
* `GetAttributeAsync<T>` from [PuppeteerSharp.Dom.CodeGen.Partial](https://github.com/urgcorp/PuppeteerSharp.Dom/blob/a29fe2cecd3c5e46fc09a74e3c050f8ea96f9b07/PuppeteerSharp.Dom/PuppeteerSharp.Dom.CodeGen.Partial.cs) - at [line 642](https://github.com/urgcorp/PuppeteerSharp.Dom/blob/a29fe2cecd3c5e46fc09a74e3c050f8ea96f9b07/PuppeteerSharp.Dom/PuppeteerSharp.Dom.CodeGen.Partial.cs#L642)

While its all syntetic there is quite a set of tests and not a single time I catch `JsonElement` being null

To check just copy solution, put breakpoints on mentioned lines and run all tests

**Comment by CaCTuCaTu4ECKuu** [comment]
I just catched an exception because result while calling `IJSHandle.EvaluateFunctionAsync` with script `(element, name) => { return element.getAttribute(name)` and name argument of `data-address` on random page, but it wasn't expected behaviour because I accidently click something and wasn't able to reproduce this, and now I'm not even sure if I closed browser windows or just did something with page or navigation that messes with it.

But I guess main question is resolved - method can return null value.
However I'm still not sure if it can return null under regular chromium state

Best I could get is JSHandle that says that it's value is null but
![image](https://github.com/user-attachments/assets/315736fd-4088-4553-82f7-ef265c4c3251)
